pkg/writers/buffered_file_writer/bufferedfilewriter.go

```diff
@@ -0,0 +1,240 @@
+// Package bufferedfilewriter provides a writer that buffers data in memory until a threshold is exceeded at
```

Comment:
No so the base branch I just introduced the interface and left everything almost as is. the base branch uses a buffer struct which is simply a wrapper around bytes.Buffer to preserve all existing functionality. This PR provides a mechanism via a feature flag to use the bufferedFileWriter instead of the default buffer. I split them up.

Reply:
i see a bunch of temp file stuff in https://github.com/trufflesecurity/trufflehog/blob/feat-tmp-file-diffs/pkg/writers/buffered_file_writer/bufferedfilewriter.go so let's have a synchronous chat to figure out what weird thing i'm doing

Reply:
Ahhh i see what you mean. I'll fix it and we can chat sync as well. Thanks for pointing that out, not sure why the files stayed in the base PR.